### ISSUE TITLE
Pyinstaller breaks tkterminal 

### ISSUE THREAD

**Issue by Moosems** [pyinstaller/pyinstaller#6736]
### Context information (for bug reports)

* Output of `pyinstaller --version`: ```410```
* Version of Python: 3.9.10 (likely on all versions)
* Platform: MacOSX
* How you installed Python: python.org/downloads
* Did you also try this on another platform? Does it work there? No, I don't have any other platforms to test on

* try the latest development version, using the following command:

```shell
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
```

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [✓] start with clean installation
  * [✓] use the latest development version
  * [✓] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [ ] Package your program in **--onedir mode**
  * [ ] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [ ] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

```
from tkinter import *
from tkterminal import *

window = Tk()
terminal = Terminal(window)
terminal.pack()
terminal.shell = True
terminal.linebar = True
terminal.run_command("cd /")
terminal.clear()
window.mainloop()
```

### Stacktrace / full error message

```
None
```

Please also see <https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs>
for more about what would use to solve the issue.

**Comment by bwoodsend** [comment]
This is the way macOS's windowed apps work I'm afraid. The _default_ set of environment variables that all `.app` applications run in (see the output of `printenv` in your frozen `tkterminal` application) is bare. In particular, `PATH` will only contain the core system paths (`/bin` and `/usr/bin` if I recall correctly). The `Terminal.app` runs a whole chain of processes and initialisation scripts. This is just from memory but I think it's:
```
Terminal.app
-> launches login
  -> Sets $HOME
  -> sources /etc/profile
  -> sources /etc/profile.d/*
  -> sources ~/.profile
  -> launches zsh (or your alternative shell if configured)
    -> sources ~/.zshrc
    -> Finally start interactively running user supplied commands
```
In short, any environment variable set in any of those initialisation steps/scripts will only be visible in `Terminal.app`.

That `python3` you're expecting to find is probably hiding in `/usr/local/bin` which will not be in `PATH` by default but will be added to `PATH` in either your `.zshrc` or a `.profile` so it's findable in terminal but not anywhere else. The same will be happening with `python` except that macOS provides a system-wide installation of Python 2 (in `/usr/bin/python` I think) which is in `PATH` so it's findable but, as it's not the Python you were hoping to find, it doesn't contain any 3rd party libraries - hence the `ImportError`s.

**Comment by Moosems** [comment]
@bwoodsend How could I make a change to the tkterminal code to reference /usr/local/bin? Is there a way to do this?

**Comment by rokm** [comment]
You could add it to `PATH` at the start of your `tkterminal` program if it is not already in there (check `os.environ['PATH']`).

But if you expect to have a `python3` available /usr/local/bin and have it custom packages installed (in order to run some unfrozen script), I guess the question is why bother freezing the `tkterminal` program in the first place...
